assign_rights/test_helpers.py

```diff
@@ -58,79 +58,31 @@ def set_dates():
-            copyright_status=random.choice(["copyrighted", "public domain", "unknown"]),
-            jurisdiction="us",
-            determination_date=random_date(10, 0),
-            note=random_string(),
-            start_date=date_info[0],
-            end_date=date_info[1],
-            start_date_period=date_info[2],
-            end_date_period=date_info[3],
-            end_date_open=date_info[4],
-        )
-
-    def other_shell():
-        date_info = set_dates()
-        RightsShell.objects.create(
-            rights_basis=random.choice(["policy", "donor"]),
-            determination_date=random_date(10, 0),
-            note=random_string(),
-            start_date=date_info[0],
-            end_date=date_info[1],
-            start_date_period=date_info[2],
-            end_date_period=date_info[3],
-            end_date_open=date_info[4],
-        )
-
-    def statute_shell():
-        date_info = set_dates()
-        RightsShell.objects.create(
-            rights_basis="statute",
-            jurisdiction="us",
-            determination_date=random_date(10, 0),
-            note=random_string(),
-            start_date=date_info[0],
-            end_date=date_info[1],
-            start_date_period=date_info[2],
-            end_date_period=date_info[3],
-            end_date_open=date_info[4],
-            statute_citation=random_string(),
-        )
-
-    def license_shell():
-        date_info = set_dates()
-        RightsShell.objects.create(
-            rights_basis="license",
-            jurisdiction="us",
-            determination_date=random_date(10, 0),
-            note=random_string(),
-            start_date=date_info[0],
-            end_date=date_info[1],
-            start_date_period=date_info[2],
-            end_date_period=date_info[3],
-            end_date_open=date_info[4],
-            license_terms=random_string(),
-        )
-
-    shell_types = [copyright_shell, other_shell, statute_shell, license_shell]
     for x in range(count):
-        random.choice(shell_types)()
+        start_date, end_date, start_date_period, end_date_period, end_date_open = set_dates()
+        new_shell = RightsShell.objects.create(determination_date=random_date(10, 0), note=random_string(), start_date=start_date, end_date=end_date, start_date_period=start_date_period, end_date_period=end_date_period, end_date_open=end_date_open)
+        basis = random.choice(["copyright", "policy", "donor", "statute", "license"])
+        setattr(new_shell, "rights_basis", basis)
+        if basis in ["copyright", "statute"]:
+            setattr(new_shell, "jurisdiction", "us")
+        if basis == "copyright":
+            setattr(new_shell, "copyright_status", random.choice(["copyrighted", "public domain", "unknown"]))
+        if basis == "statute":
+            setattr(new_shell, "statute_citation", random_string())
+        if basis == "license":
+            setattr(new_shell, "license_terms", random_string())
```

Comment:
Cool, this is a lot more terse, and I think makes more sense (at least to me).

A couple of things, just from a quick eye test:
- I would suggest instantiating (but not saving) the RightsShell in line 63. So, `new_shell = RightsShell(determination_date...)` and then calling `new_rights.save()` after you've added all the necessary fields. I'm not sure what you have here will actually persist those fields to the database.
- I'm not sure you need to use setattr - I think you can just do `new_shell.rights_basis = basis`, for example